plugins/polio/models.py

```diff
@@ -174,12 +176,31 @@ class RoundDateHistoryEntry(models.Model):
     previous_ended_at = models.DateField(null=True, blank=True)
     started_at = models.DateField(null=True, blank=True)
     ended_at = models.DateField(null=True, blank=True)
+    # Deprecated.
     reason = models.CharField(null=True, blank=True, choices=DelayReasons.choices, max_length=200)
+    reason_for_delay = models.ForeignKey(
+        "ReasonForDelay", on_delete=models.PROTECT, null=True, blank=True, related_name="round_history_entries"
+    )
     round = models.ForeignKey("Round", on_delete=models.CASCADE, related_name="datelogs", null=True, blank=True)
     modified_by = models.ForeignKey("auth.User", on_delete=models.PROTECT, null=True, blank=True)
     created_at = models.DateTimeField(auto_now_add=True)
 
 
+class ReasonForDelay(SoftDeletableModel):
+    name = TranslatedField(models.CharField(_("name"), max_length=200), {"fr": {"blank": True, "null": True}})
+    key_name = models.CharField(null=True, blank=True, max_length=200, validators=[RegexValidator(r"^[A-Z_]+$")])
```

Comment:
Do we really need this `key_name` field?
Why not just a unique `name` by account?
Unless I missed something, I think the code could be simpler without it.

Otherwise:

```suggestion
    key_name = models.CharField(blank=True, max_length=200, validators=[RegexValidator(r"^[A-Z_]+$")])
```

Reply:
Yeah, for 3 reasons:
- The powerBi dashboards rely on them, at least for the moment
- I need to filter out `INITIAL_DATA` in the front-end, and it seems more reliable to do it this way than based on name (which is editable) or `id`
- If someone changes `INITIAL_DATA` to "Yabadabadoo", we kinda get a hint what it was rezally supposed to be. this argument is not so strong, given the presence of the logs. Still, it kinda locks the original intent.